### PR TITLE
Add custom xunit attributes

### LINKED ISSUE #1052: Add a custom Fact attribute that set the Skip value when needed
> Why did you add this condition? Was it to stop the IDE from trying to run the test? The WCF client repo has a custom XUnit test runner with a lot of nice capabilities like forcing a crash dump with the test freezes. But I think that's overkill here. I found a nice approach in the [blog post here](https://josephwoodward.co.uk/2019/01/skipping-xunit-tests-based-on-runtime-conditions) where a custom attribute that derives from FactAttribute can be used. We could replace the Trait (and the command line arguments in the pipeline yml files) with a custom attribute which sets the Skip value.

_Originally posted by @mconnew in https://github.com/CoreWCF/CoreWCF/pull/1033#discussion_r1159158385_

### PR DESCRIPTION
**Pull request by g7ed6e**
Resolves #1052

### CODE REVIEW

File: src/CoreWCF.Http/tests/HttpSysTests.cs [hunk 1]
```diff
@@ -24,8 +24,7 @@ public HttpSysTests(ITestOutputHelper output)
             _output = output;
         }
 
-        [Fact]
-        [Trait("Category", "WindowsOnly")]
+        [WindowsNetCoreOnlyFact]
 #if NET5_0_OR_GREATER
```
Comment (mconnew) [inline src/CoreWCF.Http/tests/HttpSysTests.cs:28]: I believe this one should be WindowsOnly. The #if around NET5_0 was for the benefit of the SupportedOSPlatform attribute.. The UseHttpSys api has this attribute, which then means it's needed to be added to CreateHttpSysBuilder. Without these attributes, you get compiler warnings as the test builds target net6.0 and not net6.0-windows. These attributes aren't available for .NET Framework which is why there's the conditional compilation around the attribute. The test runs just fine on .NET Framework though.

File: src/CoreWCF.Http/tests/SimpleWSHTTPTest.cs [hunk 2]
```diff
@@ -230,14 +227,9 @@ internal void WSHttpRequestImpersonateWithKestrel()
                 ((IChannel)channel).Close();
             }
         }
-#endif
 
-        [Fact]
+        [WindowsNetCoreOnlyFact]
         [Description("transport-security-with-windows-authentication-httpsys")]
```
Comment (mconnew) [inline src/CoreWCF.Http/tests/SimpleWSHTTPTest.cs:232]: I believe this should be WindowsOnly for the same reasons as previously commented.

File: src/CoreWCF.Http/tests/SimpleWSHTTPTest.cs [hunk 3]
```diff
@@ -286,12 +277,8 @@ public void WSHttpRequestImpersonateFailsWithoutAuthentication()
             }
         }
 
-        [Fact]
+        [WindowsNetCoreOnlyFact]
         [Description("no-security-with-an-anonymous-client-using-impersonation-httpsys")]
```
Comment (mconnew) [inline src/CoreWCF.Http/tests/SimpleWSHTTPTest.cs:281]: WindowsOnly, should work on .NET Framework

File: src/CoreWCF.NetTcp/tests/NetTcpHostedServiceTests.cs [hunk 4]
```diff
@@ -21,8 +21,7 @@ public NetTcpHostedServiceTests(ITestOutputHelper output)
             _output = output;
         }
 
-        [Fact]
-        [Trait("Category", "WindowsOnly")]  // HttpSys not supported on Linux
+        [WindowsNetCoreOnlyFact]  // HttpSys not supported on Linux
 #if NET5_0_OR_GREATER
```
Comment (mconnew) [inline src/CoreWCF.NetTcp/tests/NetTcpHostedServiceTests.cs:25]: WindowsOnly

**Comment by mconnew** [comment]
Once you've fixed the misattributed tests, everything else looks good so feel free to merge.

File: src/CoreWCF.Http/tests/HttpSysTests.cs (hunk 1 above)
Comment (g7ed6e) [reply src/CoreWCF.Http/tests/HttpSysTests.cs:28]: done, thank you for the explanation.

File: src/CoreWCF.Http/tests/SimpleWSHTTPTest.cs (hunk 2 above)
Comment (g7ed6e) [reply src/CoreWCF.Http/tests/SimpleWSHTTPTest.cs:232]: done

File: src/CoreWCF.Http/tests/SimpleWSHTTPTest.cs (hunk 3 above)
Comment (g7ed6e) [reply src/CoreWCF.Http/tests/SimpleWSHTTPTest.cs:284]: done

File: src/CoreWCF.NetTcp/tests/NetTcpHostedServiceTests.cs (hunk 4 above)
Comment (g7ed6e) [reply src/CoreWCF.NetTcp/tests/NetTcpHostedServiceTests.cs:26]: done

**Pull request merged**